File: 0049-group-anagrams/0049-group-anagrams.cpp
class Solution {
public:
    vector<vector<string>> groupAnagrams(vector<string>& strs) {
        unordered_map<string, vector<string>> grp_angrm_map;

        for (auto& word : strs) {
            string sorted_word = word;
            sort(sorted_word.begin(), sorted_word.end());
            grp_angrm_map[sorted_word].push_back(move(word));
        }

        vector<vector<string>> grp_angrm_list;
        for (auto& key_val : grp_angrm_map) {
            grp_angrm_list.push_back(move(key_val.second));
        }

        return grp_angrm_list;
    }
};

File: 0049-group-anagrams/0049-group-anagrams.py
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        grp_angrm_map = defaultdict(list)

        for word in strs:
            sorted_word = ''.join(sorted(word))
            grp_angrm_map[sorted_word].append(word)
        
        grp_angrm_list = []

        for val in grp_angrm_map.values():
            grp_angrm_list.append(val)

        return grp_angrm_list

        